symfony/plugins/orangehrmTimePlugin/Api/CopyProjectActivityAPI.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Api;
+
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\RecordNotFoundException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Project;
+use OrangeHRM\Time\Api\Model\CopyActivityModel;
+use OrangeHRM\Time\Dto\CopyActivityField;
+use OrangeHRM\Time\Exception\ProjectServiceException;
+use OrangeHRM\Time\Traits\Service\ProjectServiceTrait;
+
+class CopyProjectActivityAPI extends Endpoint implements CollectionEndpoint
+{
+    use ProjectServiceTrait;
+
+    public const PARAMETER_FROM_PROJECT_ID = 'fromProjectId';
+    public const PARAMETER_TO_PROJECT_ID = 'toProjectId';
+    public const PARAMETER_ACTIVITY_IDS = 'activityIds';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        list($toProjectId, $fromProjectId) = $this->getUrlAttributes();
+        $this->checkProjectAvailability($toProjectId, $fromProjectId);
+
+        $copyActivityField = new CopyActivityField();
+        $copyActivityField->setFromProjectId($fromProjectId);
+        $copyActivityField->setToProjectId($toProjectId);
```

Comment:
don't see pagination handled. And seems deviated from the usual pattern. you moved whole logic to Model

Reply:
done